source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -116,8 +116,8 @@ Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMa
   const bool isCompressible = isEnabledAndContentLengthBigEnough && isContentTypeAllowed(headers) &&
                               !hasCacheControlNoTransform(headers) && isEtagAllowed(headers) &&
                               !headers.getInline(content_encoding_handle.handle());
-  if (!end_stream && isEnabledAndContentLengthBigEnough && isAcceptEncodingAllowed(headers) &&
-      isCompressible && isTransferEncodingAllowed(headers)) {
+  if (!end_stream && isCompressible && isAcceptEncodingAllowed(headers) &&
```

Comment:
Unfortunately `isAcceptEncodingAllowed()` isn't just a predicate, it produces side effects, i.e. updates stats, and needs to be called even if the request isn't compressible.

Would be nice to make the code clearer though.